src/control/server/config/server.go

```diff
@@ -433,8 +434,99 @@ func getAccessPointAddrWithPort(log logging.Logger, addr string, portDefault int
 	return addr, nil
 }
 
+// Calculate hugepages based on total target count if not using nvme.
+func hugepageSetCount(log logging.Logger, cfgTargetCount, sysXSCount, hugepageSize int, cfg *Server) error {
+	if cfgTargetCount <= 0 {
+		return nil // no nvme, no hugepages required
+	}
+
+	if cfg.DisableHugepages {
+		return FaultConfigHugepagesDisabled
+	}
+
+	// Calculate minimum number of hugepages for all configured engines.
+	minHugepages, err := common.CalcMinHugepages(hugepageSize, cfgTargetCount+sysXSCount)
+	if err != nil {
+		return err
+	}
+
+	// If the config doesn't specify hugepages, use the minimum. Otherwise, validate
+	// that the configured amount is sufficient.
+	if cfg.NrHugepages == 0 {
+		var msgSysXS string
+		if sysXSCount > 0 {
+			msgSysXS = fmt.Sprintf(" and %d sys-xstreams", sysXSCount)
+		}
+		log.Debugf("calculated nr_hugepages: %d for %d targets%s", minHugepages,
+			cfgTargetCount, msgSysXS)
+		cfg.NrHugepages = minHugepages
+	}
+
+	if cfg.NrHugepages < minHugepages {
+		return FaultConfigInsufficientHugepages(minHugepages, cfg.NrHugepages)
+	}
+
+	return nil
+}
+
+// Calculate thresholds for scm tmpfs size using nr hugepages from config and total memory
+// as reported by /proc/meminfo. Validate configured values or assign if not already set.
+func scmSetSize(log logging.Logger, mi *common.MemInfo, cfg *Server) error {
+	if len(cfg.Engines) == 0 {
+		return nil // no engines
+	}
+
+	// Create the same size scm for each engine.
+	scmCfgs := cfg.Engines[0].Storage.Tiers.ScmConfigs()
+
+	if len(scmCfgs) == 0 || scmCfgs[0].Class != storage.ClassRam {
+		return nil // no tmpfs to size
+	}
+
+	// Convert available memory from kib to bytes.
+	memTotal := uint64(mi.MemTotalKiB * humanize.KiByte)
+
+	// Calculate assigned hugepage memory in bytes.
+	memHuge := uint64(cfg.NrHugepages * mi.HugepageSizeKiB * humanize.KiByte)
+
+	// Pass 0 for sys and engine reservations to use default values.
+	// TODO: pass system reservation parameter from config
+	scmSize, err := common.CalcScmSize(log, memTotal, memHuge, 0, 0, len(cfg.Engines))
```

Comment:
As an aside, it seems a bit weird to have this stuff in the `common` package. I know that it's there because that's where the `MemInfo` struct wound up, but it seems like it'd be better in the `storage` package. e.g. `storage.CalcRamdiskSize()`, etc.

Reply:
done.